python/ray/train/_internal/data_config.py

```diff
@@ -20,6 +20,7 @@ def __init__(
         self,
         datasets_to_split: Union[Literal["all"], List[str]] = "all",
         execution_options: Optional[ExecutionOptions] = None,
+        convert_to_data_iterator: bool = True,
```

Comment:
Nit: This seems unclear for the people don't know about what data_iterator refers to. Consider rename it to `streaming_execution=True` or `materialize=False`?

Reply:
Update: we're gonna have the user call [`DataIterator.materialize`](https://github.com/ray-project/ray/pull/43210) instead.